### PR TITLE
Add clear()

### LINKED ISSUE #648: clear() method
A DB method for removing all the keys and values. Currently doing it like:
```javascript
db.createKeyStream().on("data", key => {
    db.del(key);
});
```

### PR DESCRIPTION
**Pull request by vweevers**
Closes #648.

~~Blocked by https://github.com/Level/deferred-leveldown/pull/72 and https://github.com/Level/encoding-down/pull/89.~~

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -446,6 +464,7 @@ const main = async () => {
 | `put`     | Key has been updated        | `key, value` (any)   |
 | `del`     | Key has been deleted        | `key` (any)          |
 | `batch`   | Batch has executed          | `operations` (array) |
+| `clear`   | Entries were deleted        | `options` (object)   |
```
Comment (vweevers) [inline README.md]: Not 100% sure about the argument (`options`). I figured a listener might want to inspect the range options, to find out what was deleted. But that information is not exact (in contrast with the `put`, `del` and `batch` events that tell you exactly which keys were modified).

**Comment by vweevers** [comment]
@ralphtheninja Before we release this I want to think about adding a way to feature-detect it.

A module like [`noffle/kappa-view-level`](https://github.com/noffle/kappa-view-level) (used in `cabal-core`) may want to use `db.clear()` for its [`clearIndex()`](https://github.com/noffle/kappa-view-level/blob/0665b631669b34b3f11935f9611a8118e5c73641/index.js#L47-L68). If `cabal-core` were to switch to `level-mem` (instead of the old `memdb` that it currently uses) then the `db` it passes into `kappa-view-level` would indeed have a `clear()` method. Problem is, both these modules support passing in a custom db.

That db may be `levelup` with `clear()`, but its underlying db may _not_ have `clear()`. So a check like `typeof db.clear === 'function'` would not suffice. Even worse, the db may be `subleveldown` with the default `clear()` of `abstract-leveldown` and delete keys outside of the sublevel range :(

**Comment by ralphtheninja** [comment]
Maybe we could start by adding a `clear()` to `subleveldown` to prevent from falling back to default behavior? But I guess the damage is already done there if it already falls back to using default `clear()`.

**Comment by vweevers** [comment]
Yeah, we could make a patch release on `subleveldown` to either pin `abstract-leveldown` or add a noop `_clear()`. No one is using `clear()` yet and it isn't exposed on the outer `levelup` instance yet so we have time.

**Comment by vweevers** [comment]
It's time for manifests (again), I think. https://github.com/Level/community/issues/42#issuecomment-529196502

**Comment by ralphtheninja** [comment]
> It's time for manifests (again), I think. [Level/community#42 (comment)](https://github.com/Level/community/issues/42#issuecomment-529196502)

I was thinking manifest just now :)

**Comment by vweevers** [comment]
I propose the following timeline:

1. Patch `subleveldown`
2. Release `levelup` minor with `clear()` (floats into `level` and friends)
3. Start working on manifests, as the "next big thing". In parallel, add `clear()` to `subleveldown`

**Pull request merged**